pipeline/03-evaluate.R

```diff
@@ -65,59 +65,56 @@ assessment_data_pin <- read_parquet(paths$output$assessment_pin$local) %>%
 # Function to take either test set results or assessment results and generate
 # aggregate performance statistics for different levels of geography
 gen_agg_stats <- function(data, truth, estimate, bldg_sqft,
-                          rsn_col, rsf_col, triad, geography, class, col_dict) {
+                          rsn_col, rsf_col, triad, geography,
+                          class, col_dict, min_n) {
+  # Helper function to return NA when sale sample size is too small
+  gte_n <- \(n_sales, min_n, fn) ifelse(sum(!is.na(n_sales)) >= min_n, fn, NA)
+
   # List of summary stat/performance functions applied within summarize() below
   # Each function is listed on the right while the name of the function is on
   # the left
   rs_fns_list <- list(
-    cod_no_sop = ~ ifelse(sum(!is.na(.y)) > 1, cod(.x / .y, na.rm = TRUE), NA),
-    prd_no_sop = ~ ifelse(sum(!is.na(.y)) > 1, prd(.x, .y, na.rm = TRUE), NA),
-    prb_no_sop = ~ ifelse(sum(!is.na(.y)) > 1, prb(.x, .y, na.rm = TRUE), NA),
-    mki_no_sop = ~ ifelse(sum(!is.na(.y)) > 1, mki(.x, .y, na.rm = TRUE), NA),
-    cod = ~ ifelse(
-      sum(!is.na(.y)) > 33,
-      list(ccao_cod(.x / .y, na.rm = TRUE)),
-      NA
-    ),
-    prd = ~ ifelse(
-      sum(!is.na(.y)) > 33,
-      list(ccao_prd(.x, .y, na.rm = TRUE)),
-      NA
-    ),
-    prb = ~ ifelse(
-      sum(!is.na(.y)) > 33,
-      list(ccao_prb(.x, .y, na.rm = TRUE)),
-      NA
-    )
+    cod_no_sop = \(x, y) gte_n(y, 2, cod(x / y, na.rm = TRUE)),
```

Comment:
This is just a rewrite of the same function but using the new `\(x) ...` anonymous function syntax.